evennia/objects/objects.py

```diff
@@ -672,6 +672,55 @@ def for_contents(self, func, exclude=None, **kwargs):
         for obj in contents:
             func(obj, **kwargs)
 
+    def receive_action_msg(self, from_obj, message, mapping, raise_funcparse_errors=False, **kwargs):
+        """
+        Receive and process an action message from another object. Refer to msg_contents for the docs.
+
+        This is a hook so that it can be overridden for customization of receiving action messages.
+
+        An example, would be if you want to relay the action message to a different object and format it
+        differently for them.
+        """
+        if is_iter(message):
+            text, outkwargs = message
+        else:
+            text = message
+            outkwargs = {}
+
+        outmessage = _MSG_CONTENTS_PARSER.parse(
+            text,
+            raise_errors=raise_funcparse_errors,
+            return_string=True,
+            caller=from_obj,
+            receiver=self,
+            mapping=mapping,
+        )
+
+        # director-stance replacements
+        outmessage = outmessage.format_map(
+            {
+                key: obj.get_display_name(looker=self)
+                if hasattr(obj, "get_display_name")
+                else str(obj)
+                for key, obj in mapping.items()
+            }
+        )
+
+        self.msg(text=(outmessage, outkwargs), from_obj=from_obj, **kwargs)
+
+    def send_action_msg(self, message, receiver, mapping=None, raise_funcparse_errors=False, **kwargs):
```

Comment:
This is not a useful method name; let's not introduce a new `send_` and `action` nomenclature. Since you can call this in order to send an actor-stance aware message, maybe it's best to note that in the name, like `msg_with_actor_stance`